### PR TITLE
Richtext

### PR DESCRIPTION
**Pull request by frabcus**

### CODE REVIEW

File: messytables/excel.py [hunk 1]
```diff
@@ -138,10 +138,51 @@ def properties(self):
...
 class XLSProperties(CoreProperties):
     KEYS = ['bold', 'size', 'italic', 'font_name', 'strikeout', 'underline',
-            'font_colour', 'background_colour', 'any_border', 'all_border']
+            'font_colour', 'background_colour', 'any_border', 'all_border',
+            'richtext']
     def __init__(self, cell):
-        self.xf = cell.sheet.book.xf_list[cell.xlrd_cell.xf_index]
-        self.font = cell.sheet.book.font_list[self.xf.font_index]
+        #self.xf = cell.sheet.book.xf_list[cell.xlrd_cell.xf_index]
+        #self.font = cell.sheet.book.font_list[self.xf.font_index]
```
Comment (frabcus) [inline messytables/excel.py]: Be bold, delete these two lines!

File: messytables/excel.py [hunk 2]
```diff
@@ -138,10 +138,51 @@ def properties(self):
...
+    def xf(self):
+        return self.cell.sheet.book.xf_list[self.cell.xlrd_cell.xf_index]
+
+    @property
+    def font(self):
+        return self.cell.sheet.book.font_list[self.xf.font_index]
+
+    @property
+    def rich(self):
+        return self.cell.sheet.rich_text_runlist_map.get(self.cell.xlrd_pos, None)
```
Comment (frabcus) [inline messytables/excel.py]: Should have a docstring saying what it returns, and maybe linking to https://secure.simplistix.co.uk/svn/xlrd/trunk/xlrd/doc/xlrd.html?p=4966#sheet.Sheet.rich_text_runlist_map-attribute

File: messytables/excel.py [hunk 3]
```diff
@@ -138,10 +138,51 @@ def properties(self):
...
+
+    def raw_span(self, always=False):
```
Comment (frabcus) [inline messytables/excel.py]: Comment or docstring referencing https://secure.simplistix.co.uk/svn/xlrd/trunk/xlrd/doc/xlrd.html?p=4966#sheet.Sheet.merged_cells-attribute ?

File: messytables/excel.py [hunk 4]
```diff
@@ -138,10 +138,51 @@ def properties(self):
...
+        "return the bounding box of the cells it's part of."
+        row, col = self.cell.xlrd_pos
+        for box in self.cell.sheet.merged_cells:
+            rlo, rhi, clo, chi = box
+            if row >= rlo and row <= rhi and col >= clo and col <= chi:
+                return box
+        if always:
+            print self.cell.xlrd_pos, box
```
Comment (frabcus) [inline messytables/excel.py]: Do we want this debugging around? If so should print with a prefix, or else nobody will be able to grep for it.

File: messytables/excel.py [hunk 5]
```diff
@@ -138,10 +138,51 @@ def properties(self):
...
+            return (row, row, col, col)
+        else:
+            print self.cell.xlrd_pos, None
+            return None
+
+    @property
+    def topleft(self):
```
Comment (frabcus) [inline messytables/excel.py]: Should this be called is_topleft? Perhaps not. Maybe just say in docstring that it returns a boolean as to whether topleft.

File: test/test_properties.py [hunk 6]
```diff
@@ -51,6 +51,24 @@ def test_properties_implements_get(self):
...
 
 
+class TestExcelSpanRich(unittest.TestCase):
+    @classmethod
+    def setUpClass(cls):
+        cls.xls = any_tableset(horror_fobj('span_rich.xls'), extension='xls')
+        cls.table = list(list(cls.xls.tables)[0])
+
+    def test_basic_rich(self):
+        assert self.table[4][1].properties['is_rich']
```
Comment (frabcus) [inline test/test_properties.py]: is rich now

Comment (frabcus) [inline test/test_properties.py]: change to .is_rich

File: messytables/excel.py (hunk 5 above)
Comment (scraperdragon) [inline messytables/excel.py]: It's part of the contract of properties that topleft is called that... we'd need to change throughout.

**Pull request merged**